<a href="https://colab.research.google.com/github/RAK789266/http-localhost-8888-notebooks-Untitled13.ipynb-kernel_name-python3/blob/main/UNIFIED_MENTOR.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd

# Load the dataset
df = pd.read_csv("/content/greendestination (1) (1).csv")

In [3]:
# Attrition rate: % of people who have left
attrition_rate = df['Attrition'].value_counts(normalize=True).get('Yes', 0) * 100
print(attrition_rate)

16.122448979591837


The output means that based on our dataset, 16.12% of the employees have left the company. This is our employee attrition rate.

**Purpose**: Calculates the percentage of people who have left ("Attrition" marked as 'Yes').

**Steps:**

df['Attrition'].value_counts(normalize=True) calculates the proportion for each value ("Yes" or "No") in the "Attrition" column.

.get('Yes', 0) fetches the proportion of "Yes".

Multiplied by 100 for percentage.

Output: Prints the % of people who have left.

In [4]:
# Evaluate if factors play a part in attrition
from scipy.stats import ttest_ind
# Age
age_left = df[df['Attrition'] == 'Yes']['Age']
age_stayed = df[df['Attrition'] == 'No']['Age']
t_age, p_age = ttest_ind(age_left, age_stayed, nan_policy='omit')
print(t_age)
print(p_age)

-6.178663835307217
8.356308021103587e-10


**Purpose:**Checks if there is a significant difference in age between people who left versus those who stayed.

**Steps:**

age_left: Ages of those who left.

age_stayed: Ages of those who stayed.

ttest_ind: Performs an independent T-test comparing the two groups.

Returns:

t_age: Test statistic.

p_age: p-value (significance).

nan_policy='omit' ignores missing values.

A low p-value (<0.05) suggests age is significantly different between the two groups.



In [5]:
# Monthly Income
income_left = df[df['Attrition'] == 'Yes']['MonthlyIncome']
income_stayed = df[df['Attrition'] == 'No']['MonthlyIncome']
t_income, p_income = ttest_ind(income_left, income_stayed, nan_policy='omit')
print(t_income)
print(p_income)

-6.203935765608938
7.147363985353811e-10


**Purpose:** Checks if there is a significant difference in monthly income between people who left versus those who stayed.

**Steps:**

income_left: Monthly income for those who left.

income_stayed: Monthly income for those who stayed.

ttest_ind: Performs T-test to compare the two groups.

In [6]:
# Years at company
years_left = df[df['Attrition'] == 'Yes']['YearsAtCompany']
years_stayed = df[df['Attrition'] == 'No']['YearsAtCompany']
t_years, p_years = ttest_ind(years_left, years_stayed, nan_policy='omit')
print(t_years)
print(p_years)

-5.1963086670254235
2.3188716103863036e-07


**T-statistic (t_years)** = -5.196

This value measures the size of the difference between the two groups relative to the variation in the data.

Negative sign indicates that the mean of the first group (years_left) is smaller than the mean of the second group (years_stayed). In simple terms, employees who left had, on average, a shorter tenure than employees who stayed.

**p-value (p_years)** = 2.318...e-07

This is the most important value for the conclusion. It's written in scientific notation and is equal to 0.0000002318.

The p-value represents the probability of observing such a difference (or an even larger one) between the two groups by pure chance, assuming there was no real difference between them.



